java.lang.NullPointerException: Cannot invoke "cs451.Message.Message.getReceiverId()" because "m" is null
	at cs451.links.StubbornLinks.lambda$sendMessagesToBeSent$1(StubbornLinks.java:94)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at cs451.links.StubbornLinks.sendMessagesToBeSent(StubbornLinks.java:92)
	at cs451.links.StubbornLinks.lambda$new$0(StubbornLinks.java:71)
	at java.base/java.lang.Thread.run(Thread.java:833)
java.lang.NullPointerException: Cannot invoke "cs451.Message.Message.getReceiverId()" because "m" is null
	at cs451.links.StubbornLinks.lambda$sendMessagesToBeSent$1(StubbornLinks.java:94)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at cs451.links.StubbornLinks.sendMessagesToBeSent(StubbornLinks.java:92)
	at cs451.links.StubbornLinks.lambda$new$0(StubbornLinks.java:71)
	at java.base/java.lang.Thread.run(Thread.java:833)
java.lang.NullPointerException
java.lang.NullPointerException
java.net.SocketException: Socket closed
	at java.base/sun.nio.ch.DatagramSocketAdaptor.receive(DatagramSocketAdaptor.java:252)
	at java.base/java.net.DatagramSocket.receive(DatagramSocket.java:701)
	at cs451.udp.UDPReceiver.run(UDPReceiver.java:46)
Caused by: java.nio.channels.AsynchronousCloseException
	at java.base/sun.nio.ch.DatagramChannelImpl.endRead(DatagramChannelImpl.java:523)
	at java.base/sun.nio.ch.DatagramChannelImpl.trustedBlockingReceive(DatagramChannelImpl.java:677)
	at java.base/sun.nio.ch.DatagramChannelImpl.blockingReceive(DatagramChannelImpl.java:635)
	at java.base/sun.nio.ch.DatagramSocketAdaptor.receive(DatagramSocketAdaptor.java:240)
	... 2 more
java.net.SocketException: Socket closed
	at java.base/sun.nio.ch.DatagramSocketAdaptor.send(DatagramSocketAdaptor.java:222)
	at java.base/java.net.DatagramSocket.send(DatagramSocket.java:665)
	at cs451.udp.UDPBulkSender.run(UDPBulkSender.java:34)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.DatagramChannelImpl.ensureOpen(DatagramChannelImpl.java:267)
	at java.base/sun.nio.ch.DatagramChannelImpl.blockingSend(DatagramChannelImpl.java:850)
	at java.base/sun.nio.ch.DatagramSocketAdaptor.send(DatagramSocketAdaptor.java:218)
	... 7 more
